pkg/io/storage/storage.go

```diff
@@ -0,0 +1,90 @@
+/*
+Package storage provides functions for storing and retrieving data.
+from json file
+*/
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type fileVault struct {
+	storage map[string][]byte
+	path    string
+	file    *os.File
+}
+
+func NewFileVault(path string) (*fileVault, error) {
+	var storage = make(map[string][]byte)
+
+	file, err := os.Open(filepath.Clean(path))
+	if os.IsNotExist(err) {
+		file, err = os.Create(path)
+		if err != nil {
+			return nil, fmt.Errorf("unable to open or create file: %w", err)
+		}
+	} else {
```

Comment:
It is not enough to check err on only one type of it. In case if there is an error but of another kind it will not be handled.
```suggestion
	} else if err != nil {
	// ...handle other error types
	}
	// handle success on file opening
```